### ISSUE TITLE
6.036_Machine_Learning

### ISSUE THREAD

**Issue by jekadwell** [TangJiahui/6.036_Machine_Learning#1]
Tangjiahui,
Thank you very much for posting this code for MIT 6.036.   I am writing to point our an error in the xo_more theta and theta_0 in your post.  

The perceptron actually requires 554 iterations of T to find the correct separator.  Your code has a limit of 100.  The theta and theta_0 you report is the last one generated, but not the correct separator.  I have attached a graph of your separator (Screenshot xo_more).   

Again, thank you very much for posting this code.  I would be lost without it.  You are a far, far better coder and Machine Learning student than me.  

![Screenshot xo_more](https://user-images.githubusercontent.com/44558372/225454536-5872cd30-739b-461f-8d46-f0ce997f302d.jpg)
![Screenshot xo_more_2](https://user-images.githubusercontent.com/44558372/225454565-35b696a7-653e-4602-b3bf-6b7b217a86fc.jpg)
![Screenshot xo_more_3](https://user-images.githubusercontent.com/44558372/225454599-8fe3272c-2e7f-44b1-b9fc-861ca8848931.jpg)

**Comment by tangjiahui** [comment]
Good catch! Thank you for pointing this out @jekadwell ! I think I set limit of iterations and it comes to an early stop. Your visualization with separators make more sense. Thanks for looking into this and hope you have a great time playing around with my imperfect codes!
